### PR TITLE
feat: Add Grounding module configuration

### PR DESCRIPTION
**Pull request by deekshas8**
## Context

AI/gen-ai-hub-sdk-js-backlog#161.

Add grounding module configuration. I based the changes off of the spec [here](https://github.tools.sap/AI/llm-orchestration/blob/v0.36.1/src/spec/api.yaml). However, its not released to api hub yet. This PR also does not reflect all the changes from this version of the spec, only grounding.

We could:
1. Review and merge this and then once the actual released version is available, re-generate in case there are still any changes
2. Hold this PR until the release version is available (in case there are last minute changes in the spec, not sure how likely this is though)

## Definition of Done

- [x] Code is tested (Unit, E2E)
- [x] Error handling created / updated & covered by the tests above
- [ ] Documentation updated
- [ ] (Optional) Aligned changes with the Java SDK
- [ ] (Optional) Release notes updated

### CODE REVIEW

File: sample-code/src/orchestration.ts
Comment (jjtang1985) [inline sample-code/src/orchestration.ts:231]: Maybe we can consider some convenience API as a follow up for e.g., the filter.

File: packages/orchestration/src/client/api/schema/grounding-module-config.ts
Comment (jjtang1985) [inline packages/orchestration/src/client/api/schema/grounding-module-config.ts:10]: [q] should this be added to public API?

**Review by jjtang1985 — APPROVED** [review]
LGTM as "low level" API.
Please also add potential follow ups for:
- release notes
- documentation
- convenience API
- ...

**Review by ZhongpinWang — COMMENTED** [review]
LGTM.

[pp] But maybe we wait until the new spec is out? Because `api.yaml` is not updated and this is inconsistent. Ground may also be a potential blocker for the next release if we merge now.

File: sample-code/src/orchestration.ts
Comment (deekshas8) [reply sample-code/src/orchestration.ts:231]: https://github.tools.sap/AI/gen-ai-hub-sdk-js-backlog/issues/186

**Review by jjtang1985 — CHANGES_REQUESTED** [review]
As we are waiting for the green light from orchestration, let's block this PR.

**Review by jjtang1985 — APPROVED** [review]
Aligned with orchestration.

**Pull request merged**
